nix/update.sh

```diff
@@ -2,14 +2,17 @@
 
 set -euo pipefail
 
-if [ ! -v NIX_PROFILE ]; then
+if [ -z "$NIX_PROFILE" ]; then
```

Comment:
`-v` seems to be a modern bashism so maybe we shouldn't rely on it. We'd have to use `[ -z "${NIX_PROFILE+x}" ]` so the undefined variable setting above doesn't bite us

Reply:
Ah I was wondering why this started failing, I am on a new mac and it has `bash 3.2` need to upgrade my bash apparently. `-v` was throwing errors for me, hence the change. I'll update bash and see if that resolves it.